extension/src/popup/ducks/authServices.ts

```diff
@@ -18,20 +18,23 @@ interface ErrorMessage {
   errorMessage: string;
 }
 
-export const createAccount = createAsyncThunk(
-  "auth/createAccount",
-  async (password: string) => {
-    let res;
-
-    try {
-      res = await createAccountService(password);
-    } catch (e) {
-      console.error(e);
-    }
+export const createAccount = createAsyncThunk<
+  { publicKey: string },
+  string,
+  { rejectValue: ErrorMessage }
+>("auth/createAccount", async (password, thunkApi) => {
+  let res = { publicKey: "" };
 
-    return res;
-  },
-);
+  try {
+    res = await createAccountService(password);
+  } catch (e) {
+    console.error("Failed when creating an account: ", e.message);
+    return thunkApi.rejectWithValue({
+      errorMessage: e.message,
+    });
```

Comment:
👍  We might want to reject with the entire error, instead of just the message?

Reply:
interface for ErrorMessage is `string`. Should I change it to object?

Reply:
I don't think this needs a change right now, but we might want to if we start feeling a need for more error details.